src/gpuarray_buffer_cuda.c

```diff
@@ -71,9 +71,13 @@ cuda_context *cuda_make_ctx(CUcontext ctx, int flags) {
   if (err != CUDA_SUCCESS) {
     goto fail_stream;
   }
-  err = cuStreamCreate(&res->mem_s, CU_STREAM_NON_BLOCKING);
-  if (err != CUDA_SUCCESS) {
-    goto fail_mem_stream;
+  if (ISSET(res->flags, GA_CTX_SINGLE_STREAM)) {
+    res->mem_s = res->s;
+  } else {
+    err = cuStreamCreate(&res->mem_s, 0);
```

Comment:
Can you add a comment about that? This is the thing that woudl prbably be forgotten when you write the doc about this.